### PR TITLE
1.18.0 changelog

### PR DESCRIPTION
**Pull request by sarahalsmiller**
### Description

Changelog for 1.18.0 release

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,51 @@
...
+* connect: Replace usage of deprecated Envoy field `envoy.config.core.v3.HeaderValueOption.append`. [[GH-20078](https://github.com/hashicorp/consul/issues/20078)]
+* connect: Replace usage of deprecated Envoy fields `envoy.config.route.v3.HeaderMatcher.safe_regex_match` and `envoy.type.matcher.v3.RegexMatcher.google_re2`. [[GH-20013](https://github.com/hashicorp/consul/issues/20013)]
+* docs: add Link API documentation [[GH-20308](https://github.com/hashicorp/consul/issues/20308)]
+* resource: lowercase names enforced for v2 resources only. [[GH-19218](https://github.com/hashicorp/consul/issues/19218)]
+
+BUG FIXES:
+
+* dns: SERVFAIL when resolving not found PTR records. [[GH-20679](https://github.com/hashicorp/consul/issues/20679)]
+* raft:  Fix panic during downgrade from enterprise to oss. [[GH-19311](https://github.com/hashicorp/consul/issues/19311)]
+* raft:  Fix panic during downgrade from enterprise to oss.
```
Comment (andrewstucki) [inline CHANGELOG.md:43]: this looks like it's doubled up with the above.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,51 @@
...
+* mesh: Update Envoy versions to 1.28.1, 1.27.3, and 1.26.7 to address [CVE-2024-23324](https://github.com/envoyproxy/envoy/security/advisories/GHSA-gq3v-vvhj-96j6), [CVE-2024-23325](https://github.com/envoyproxy/envoy/security/advisories/GHSA-5m7c-mrwr-pm26), [CVE-2024-23322](https://github.com/envoyproxy/envoy/security/advisories/GHSA-6p83-mfmh-qv38), [CVE-2024-23323](https://github.com/envoyproxy/envoy/security/advisories/GHSA-x278-4w4x-r7ch), [CVE-2024-23327](https://github.com/envoyproxy/envoy/security/advisories/GHSA-4h5x-x9vh-m29j), and [CVE-2023-44487](https://github.com/envoyproxy/envoy/security/advisories/GHSA-jhv4-f7mr-xx76) [[GH-20589](https://github.com/hashicorp/consul/issues/20589)]
+* mesh: update supported envoy version 1.28.0 in addition to 1.25.11, 1.26.6, 1.27.2, 1.28.0 to address [CVE-2023-44487](https://github.com/envoyproxy/envoy/security/advisories/GHSA-jhv4-f7mr-xx76) [[GH-19879](https://github.com/hashicorp/consul/issues/19879)]
```
Comment (andrewstucki) [inline CHANGELOG.md:13]: @david-yu do we normally collapse these changelog entries? The versions seem confusing seeing all of these exploded. Also, both the bottom entry and the top entry here both talk about bumping envoy versions due to  `CVE-2023-44487`...

**Review by missylbytes — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (david-yu) [reply CHANGELOG.md:13]: Yes we normally would collapse these to at least represent the Envoy versions supported by Consul for 1.18.0.

File: CHANGELOG.md (hunk 1 above)
Comment (david-yu) [reply CHANGELOG.md:43]: Yes we should remove.

Comment (sarahalsmiller) [inline CHANGELOG.md:43]:
```suggestion
```

**Pull request merged**